### PR TITLE
Code made cleaner

### PR DESCRIPTION
**Pull request by wetda3**
1) Variables and their data types are now declared at the top instead of throughout the code
2) Tile map signal now sends a dictionary instead of several variables
3) Lots of unnecessary Vector2's where removed

Also from now on I want to put any code clean-up/fixes into this feature and if possible combine all other small cleaning/fixes with this one

### CODE REVIEW

File: Scripts/TileMap.gd [hunk 1]
```diff
@@ -1,43 +1,84 @@
...
+signal proportions(grid_data)
+
+
 #this is the size of each tile
-@export var our_tile_size = Vector2i(16,16)
+@export var our_tile_size = Vector2(16,16)
 
 #how many tiles the grid should be
-@export var our_grid_size = Vector2i(12,12)
+@export var our_grid_size = Vector2(12,12)
```
Comment (werydude) [inline Scripts/TileMap.gd:10]: Why was this changed?

File: Scripts/TileMap.gd [hunk 2]
```diff
@@ -1,43 +1,84 @@
...
-	var grid = self
-	var base_tile_size = Vector2i(16,16)
-	var base_grid_size = Vector2i(12,12)
+	var grid_data : Dictionary
+	var selected_grid_shape : String
+	var base_tile_size : Vector2
+	var base_grid_size : Vector2
+	var grid_size_in_units : Vector2
+	var tile_center_offset : Vector2
+	var center_of_grid : Vector2
```
Comment (werydude) [inline Scripts/TileMap.gd:33]: Colons should be next to the variable name; see style guide (also everyone ever)

File: Scripts/TileMap.gd [hunk 3]
```diff
@@ -1,43 +1,84 @@
...
-	grid.our_grid_size = our_grid_size
+	
+	grid_size_in_units = our_tile_size * our_grid_size
+	tile_center_offset = our_tile_size/2
+	center_of_grid = grid_size_in_units/2
 	
 	#needs to be float and vector2(not 2i) otherwise will round
-	grid.scale = Vector2(float(our_tile_size[0])/float(base_tile_size[0]),float(our_tile_size[1])/float(base_tile_size[1]))
-
+	scale = our_tile_size/base_tile_size
```
Comment (werydude) [inline Scripts/TileMap.gd:46]: Rather than put this here, you can put it in the variable declaration as: 
`var variable_name := Vector2(x,y)`
Or
`var variable_name := data`

File: Scripts/TileMap.gd [hunk 4]
```diff
@@ -1,43 +1,84 @@
...
-	var our_grid_size_y = our_grid_size[1]
 	
+	grid_data = {
+		"selected_grid_shape":selected_grid_shape,
+		"tile_size":our_tile_size,
+		"grid_size_in_tiles":our_grid_size,
+		"grid_size_in_units":grid_size_in_units,
+		"scale_of_grid":scale,
+		"tile_center_offset":tile_center_offset,
+		"center_of_grid":center_of_grid
```
Comment (werydude) [inline Scripts/TileMap.gd:68]: Should be a space after the colon

File: Scripts/TileMap.gd [hunk 5]
```diff
@@ -1,43 +1,84 @@
...
+		}
```
Comment (werydude) [inline Scripts/TileMap.gd:69]: Should be deindented

File: Scripts/TileMap.gd [hunk 6]
```diff
@@ -1,43 +1,84 @@
...
+		for y in range(our_grid_size[1]):
 			#print("x: " + str(x)," y: " + str(y))
+			#placing cells should be Vector2i not Vector2
 			set_cell(0, Vector2i(x,y), 0, Vector2i(0,0), 0)
-	
-	#emit the proportions signal with the scale of the grid
-	proportions.emit(grid.scale, our_grid_size,our_tile_size)
-	
-func _process(delta):
 	pass
```
Comment (werydude) [inline Scripts/TileMap.gd:81]: Assuming I remember correctly, instead of `pass` (which suppresses errors) add a `void` return type on the function, which indicates no return.

`func my_func() -> void`

File: Scripts/TileMap.gd [hunk 7]
```diff
@@ -1,43 +1,84 @@
...
 	
+func create_circle_grid():
+	pass
```
Comment (werydude) [inline Scripts/TileMap.gd:84]: This is a correct use of pass! Nothing doing here.

File: Camera2D.gd [hunk 8]
```diff
@@ -5,6 +5,9 @@ extends Camera2D
 
 
 
+
+
+
```
Comment (werydude) [inline Camera2D.gd:10]: Every PR has this extra space. I wonder why?

File: Camera2D.gd [hunk 9]
```diff
@@ -14,50 +17,49 @@ func _process(delta):
...
+	var diff : Vector2
+	var scale_by : int
 	
 	if Take_up_full_screen == true:
 		#if true makes grid take up full screen
 		return ratio
 		
-	
 	#make grid smaller
-	var index_diff : int = grid_size.max_axis_index()
```
Comment (werydude) [inline Camera2D.gd:45]: Assuming you moved it to the top of the file; this was more correct. You didn't need the extra separation, since you already had access to the data.

File: Camera2D.gd [hunk 10]
```diff
@@ -14,50 +17,49 @@ func _process(delta):
...
+	index_diff = grid_size.max_axis_index()
 	
 	if grid_size < screen_size:
 		#make grid bigger
 		
 		#take diffenence between grid size and screen size
-		var diff = Vector2(screen_size-grid_size)
+		diff = screen_size-grid_size
```
Comment (werydude) [inline Camera2D.gd:52]: This is an example of where declaration first was smart; the contents of the data is conditional.

File: Camera2D.gd [hunk 11]
```diff
@@ -14,50 +17,49 @@ func _process(delta):
...
 		
 		#if grid is samller than screen size get the axis in which the 
 		#difference between the grid and the sccreen is the smallest
 		index_diff = diff.min_axis_index()
 			
 		
 	#size of the camera/what size the camera should be
-	var scale_by = ratio[index_diff]
-
+	scale_by = ratio[index_diff]
```
Comment (werydude) [inline Camera2D.gd:60]: This one can also be moved up top – assuming `index_diff` is already assigned – but should be typed also:

`var scale_by: int = ratio[index_diff]`

File: Camera2D.gd [hunk 12]
```diff
@@ -14,50 +17,49 @@ func _process(delta):
...
 		screen_size,
 		ratio
 	)
 
 
 func calc_camera_zoom(grid_size,grid_center,screen_size,ratio)->Vector2:
-
+	var index_diff : int
+	var diff : Vector2
+	var scale_by : int
```
Comment (werydude) [inline Camera2D.gd:39]: Remove space before colon.

**Review by werydude — DISMISSED** [review]
Lots if minor things, no logic issues that I could see. This is called refactoring btw.

Also, there are merge conflicts. please **pull remote develop and _rebase_ your branch**

File: Scripts/TileMap.gd (hunk 2 above)
Comment (wetda3) [reply Scripts/TileMap.gd:33]: no

File: Scripts/TileMap.gd (hunk 3 above)
Comment (wetda3) [reply Scripts/TileMap.gd:46]: I didn't do that for consistency. I don't want some variables to be initialized at the top and some just to be data types. It looks cleaner.

File: Scripts/TileMap.gd (hunk 4 above)
Comment (wetda3) [reply Scripts/TileMap.gd:68]: and before

File: Scripts/TileMap.gd (hunk 1 above)
Comment (wetda3) [reply Scripts/TileMap.gd:10]: because it makes my git history look more active

File: Camera2D.gd (hunk 8 above)
Comment (wetda3) [reply Camera2D.gd:10]: I am an astronaut I like space

File: Scripts/TileMap.gd (hunk 3 above)
Comment (werydude) [reply Scripts/TileMap.gd:46]: The lack of data in an uninitialized variable tells you something. In fact, doing it in one line can help the compiler optimize it. You can separate initialized and uninitialized variables into separate blocks if you like, but keep them separate.

"Looking nice" and "useful readability" are not the same thing.


Example:

```
### Exports and Inits
@export var default_grid := Vector2i(16,16)

# this is even more type safe
@export var some_number := 4 as float

# this is also fine, but redundant
# note the underscore for privacy indication
var _private_array: Array[Int] = [1,2]

### Uninit Vars
var player_data: blob
var if_this_is_still_uninit_when_I_check_it_something_wrong: bool

```
 [](edited-with-Axolo)

**Review by werydude — APPROVED** [review]
Will fix
